File: src/03/selectionSort/selectionSort.go
package main

import (
	"fmt"
)

func selectionSort(array []int) []int {
	length := len(array)

	for i := 0; i < length-1; i++ {
		minIndex := i
		for j := i + 1; j < length; j++ {
			if array[j] < array[minIndex] {
				minIndex = j
			}
		}
		array[minIndex], array[i] = array[i], array[minIndex]
	}

	return array
}

func main() {
	arr := []int{12, 32, 52, 1, 3, 56, 74, 24, 8}
	fmt.Println(selectionSort(arr))
}

File: src/03/selectionSort/selectionSort.js
function selectionSort(array) {
    const len = array.length;

    let minIndex;
    for (let i = 0; i < len - 1; i++) {
        minIndex = i;
        for (let j = i + 1; j < len; j++) {
            if (array[j] < array[minIndex]) {
                minIndex = j;
            }
        }

        [array[i], array[minIndex]] = [array[minIndex], array[i]];
    }

    return array;
}

const array = [1, 3, 2, 6, 5, 7, 8, 9, 10, 0]
console.log(selectionSort(array))
